### PR TITLE
feat(Icon): add default value for `size` & `color` props

### LINKED ISSUE #1533: IconComponent: adding default props for size and colour
1. If we simply do `<ArrowRightIcon />`  or `<ArrowRightIcon size="medium"/>` it doesn’t display the icon. They expect explicit props for size and colour. Can we please look into adding default props for the same?
2. Also if we only give colour prop `<ArrowRightIcon color="feedback.icon.neutral.lowContrast"/>`, the size exceeds beyond `2xlarge` (attached screenshot).
3. [Slack](https://razorpay.slack.com/archives/CMQ3RBHEU/p1691681309025459)
image

### PR DESCRIPTION
**Pull request by snitin315**
Fixes #1533

### CODE REVIEW

**Review by anuraghazra — DISMISSED** [review]

File: packages/blade/src/components/Icons/useIconProps/useIconProps.tsx [hunk 1]
```diff
@@ -14,8 +14,8 @@ const iconSize = {
 } as const;
 
 function useIconProps({
-  size,
-  color,
+  size = 'medium',
+  color = 'feedback.icon.neutral.lowContrast',
```
Comment (anuraghazra) [inline packages/blade/src/components/Icons/useIconProps/useIconProps.tsx:18]: Where did we decide this default color? is it from design?

Comment (snitin315) [reply packages/blade/src/components/Icons/useIconProps/useIconProps.tsx:18]: I decided it myself. Probably we can default to `currentColor` 🤔

Comment (anuraghazra) [reply packages/blade/src/components/Icons/useIconProps/useIconProps.tsx:18]: Let's ask rastogi once, design main kya hain.

Comment (snitin315) [reply packages/blade/src/components/Icons/useIconProps/useIconProps.tsx]: Updated it to `surface.text.normal.lowContrast` as per design

**Review by kamleshchandnani — APPROVED** [review]

**Pull request merged**